### PR TITLE
Redux Adding. Async operations

### PR DESCRIPTION
**Pull request by melk0sha**
**Task 6. Redux**

All data fetches have done as actions and received from store by components.
Creating, reading, deleting and updating (CRUD operations) of films have done as redux actions.
Sorting by release date and rating have done as redux actions.
Filtering by genre has done as redux actions.

### CODE REVIEW

File: src/components/resultsSection/resultsSection.jsx [hunk 1]
```diff
@@ -56,12 +40,16 @@ const ResultsSection = ({
...
-  modalValues: modalValues,
-  defaultModalValues: modalValues,
-  onModalValuesChange: func,
-  onMovieUpdate: func
+  onMoviesByParamsGet: func
 };
 
-export { ResultsSection };
+const mapStateToProps = (state) => ({
+  movies: state.movies.movieList
```
Comment (cate-blint) [inline src/components/resultsSection/resultsSection.jsx]: you can get here only length of array as a prop as you use only length in this component

File: src/constants/actions.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// App
+export const SET_MOVIES_SORT_BY = "APP/SET_MOVIES_SORT_BY";
+
+// Movies
+export const REQUEST_MOVIES_SUCCESS = "MOVIES/REQUEST_MOVIES_SUCCESS";
```
Comment (cate-blint) [inline src/constants/actions.js:9]: you may add MOVIES/REQUEST_MOVIES_INIT to indicate that loading process is started and show spinner on that to improve user experience.

File: src/components/resultsSection/resultsSection.jsx (hunk 1 above)
Comment (melk0sha) [reply src/components/resultsSection/resultsSection.jsx]: good point!

File: src/constants/actions.js (hunk 2 above)
Comment (melk0sha) [reply src/constants/actions.js:9]: i will add a comment in the code for that and will do it later 😄

**Pull request merged**